lib/tether/tether_linux.go

```diff
@@ -140,9 +140,8 @@ func (t *tether) stopReaper() {
 	signal.Reset(syscall.SIGCHLD)
 
 	// just closing the incoming channel is not going to stop the iteration
-	// so we use done channel to signal it
-	log.Debugf("Signalling the child reaper loop")
-	close(t.done)
+	// so we use the context cancellation to signal it
+	t.cancel()
```

Comment:
`childReaper` should have its own channel (called `done` or something similar) that should stop the reaper loop only. You could shut down these side loops one by one with their own done channels (did I just make up a "task" abstraction?) when tether shuts down.